### DISCUSSION TITLE
installing additonal kernels with *just* vscode jupyter extension installed?

### DISCUSSION THREAD

**Discussion by maxandersen** [microsoft/vscode-jupyter#12510] [General]
I'm trying to grok how one can install additional kernels (especially ijava or ganymede) without requiring users to install additional jupyter/python.

vscode somewhere installs jupyter but it is not on default users path thus what are the steps to use it ?

even just getting a hint on how a vscode extension could make it possible would be great then I would happily make a vscode extension that does the necessary to make it easier to try out the java kernels.

thanks!

**Comment by DonJayamanne** [comment]
The jupyter extension in vscode installs Jupyter ONLY in rare cases, such as on Mac Arm machines or a few linux distros. 

* If you happen to be using a machine that falls into one of those categories, then yet Jupyter will be used.
To find the python envionment where Jupyter is installed you can use the command `Jupyter: Select Interpreter to Start Jupyter Server` and in that quick pick the python environment will be listed.
I.e. whatever python env is used, will have been pre-selected.
You can then find that python env and install your kernel into that 
However this is not necessary, as mentioned later below, all you need to do is find the documenation and follow those notes.

* If not, then you would need to isntall ijava/ganymede kernels using the corresponding doumentations from those kernels.

**Reply by maxandersen** [reply]
Im trying to write an extension that provides the installation. Thus Cannon rely on manual steps to do in the ui. 

Im Looking for Any api or file system locations one Can use. 

When You follow the documentation one ends ups having to install additional Python environment (at least on Mac) which is what Im trying to avoid as There is already one installed by vscode.

**Comment by DonJayamanne** [comment]
> what Im trying to avoid as There is already one installed by vscode.

VS Code does not install Python, the user installs this and choes where it is installed.
I.e. on Mac M1, we ask the user to install Python (we do not ship it with vscode not is it installed by vscode, hence there's no way to provide details of this python env as its not internal to vscode)
If you want to manually credate a kernelspec file you can do so by creating the necessary kernelspec.json file in the following directory
https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs

**Reply by maxandersen** [reply]
@DonJayamanne still - in my local environment on M1/mac I have no jupyter available but it works from inside vscode. 

So it is not the same as if I had installed jupyter myself. vscode extension asked if it should install - that's what I mean by the extension doing the install.

Anyhow - what I'm really after is if there is an approach where an vscode extension can do this setup of a kernelspec (i.e, NOT manually).

**Comment by DonJayamanne** [comment]
> Anyhow - what I'm really after is if there is an approach where an vscode extension can do this setup of a kernelspec (i.e, NOT manually).

Currently there is not automated way for the extension to create kernel specs, if you as a 3rd party extension want to create kernel specs that are visible to the Jupyter extension in vscode, then you would needc to create it in the directory mentioned in the docs
https://jupyter-client.readthedocs.io/en/stable/kernels.html#kernel-specs
